staging/src/k8s.io/cloud-provider/fake/fake.go

```diff
@@ -321,8 +321,15 @@ func (f *Cloud) InstanceMetadata(ctx context.Context, node *v1.Node) (*cloudprov
 	f.addCall("instance-metadata-by-provider-id")
 	f.addressesMux.Lock()
 	defer f.addressesMux.Unlock()
+
+	providerID := node.Spec.ProviderID
```

Comment:
@andrewsykim updated!
I think now we cover all cases:
1. if the node providerID exists, we use that
2. if the node provideID not exists, for InstancesV2 we fetch providerID from InstanceMetadata
    - if InstanceMetadata.providerID is empty, the updated node has an empty providerID
    - otherwise the updated node has the providerID like "fake://12345" set from ExtID

Reply:
The fake cloud provider update looks fine to me, but I think we want this fix in the main branch first before the cherry-pick (which was why I had mentioned the manual validation first)

Reply:
ahhh I see!
Sorry I forget to mention that I've already done the manual validation for this PR
I'm going to create a follow-up PR for the fake cloud provider and also update the unit tests and manually validate it :)